packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart

```diff
@@ -862,6 +862,55 @@ void main() {
 
       expect(visualStudio.getWindows10SDKVersion(), null);
     });
+
+    testWithoutContext('Reports unicode replacement char in used properties', () {
+      const String badUtf8 = 'Bad UTF8 \u{FFFD}';
+      final List<void Function(Map<String, dynamic>)> propertyOverrides = <void Function(Map<String, dynamic>)> [
+        (Map<String, dynamic> response) => response['installationPath'] = badUtf8,
+        (Map<String, dynamic> response) => response['displayName'] = badUtf8,
+        (Map<String, dynamic> response) => response['installationVersion'] = badUtf8,
+        (Map<String, dynamic> response) {
+          final Map<String, dynamic> catalog = Map<String, dynamic>.of(response['catalog'] as Map<String, dynamic>);
+          response['catalog'] = catalog;
+          catalog['productDisplayVersion'] = badUtf8;
+        }
+      ];
```

Comment:
Let me know if there's a cleaner way to do this!

Reply:
I'd be tempted to do this as separate small but explicit tests in a group like:

```dart
group('Correctly handles bad UTF-8 in vshwere.exe output', () {
  // See: https://github.com/flutter/flutter/issues/102451

  final VisualStudioFixture fixture = setUpVisualStudio();
  final VisualStudio visualStudio = fixture.visualStudio;


  testWithoutContext('Ignores bad UTF-8 in unused properties', () {
    ...
  });

  testWithoutContext('Throws toolExit on bad UTF-8 in installationPath', () {
    final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
      ..['installationPath'] = '\u{FFFD}';
    
    setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);

    expect(() => visualStudio.isInstalled,
        throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
  });
});
```

Since these are tests, it's fine for them to be a bit boilerplatey. Generally for tests, I'd push for simplicity and clarity over succinctness -- there are probably exceptions to that rule for extreme cases :)

Reply:
Ah interesting. Updated!